<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CreateReminderDialog</class>
 <widget class="QDialog" name="CreateReminderDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>370</width>
    <height>132</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create Reminder</string>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="textLabel">
     <property name="text">
      <string>&amp;Text:</string>
     </property>
     <property name="buddy">
      <cstring>textLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="textLineEdit">
     <property name="toolTip">
      <string>Enter a description of what to be reminded</string>
     </property>
     <property name="clearButtonEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="whenLabel">
     <property name="text">
      <string>&amp;When:</string>
     </property>
     <property name="buddy">
      <cstring>whenLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QLineEdit" name="whenLineEdit">
     <property name="toolTip">
      <string>Expression to describe when the reminder should be triggered</string>
     </property>
     <property name="whatsThis">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; text-decoration: underline;&quot;&gt;Syntax Specification:&lt;/span&gt;&lt;/p&gt;&lt;p&gt;You can enter an &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;expression&amp;gt;&lt;/span&gt; to define timepoints to remind you. An expression can be: &lt;/p&gt;&lt;p&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;conjuction &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;expression&amp;gt; ; &amp;lt;expression&amp;gt; [; &amp;lt;expression&amp;gt; …]&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;timespan &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;in &amp;lt;sequence&amp;gt; [on|at|in &amp;lt;datum&amp;gt;] [&amp;lt;time&amp;gt;]&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;loop &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;every &amp;lt;type&amp;gt; [on|at|in &amp;lt;datum&amp;gt;] [&amp;lt;time&amp;gt;] [from &amp;lt;tpoint&amp;gt; [&amp;lt;time&amp;gt;]] [until &amp;lt;tpoint&amp;gt; [&amp;lt;time&amp;gt;]]&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;timeex &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;time&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;point &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;[on|next] &amp;lt;tpoint&amp;gt; [&amp;lt;time&amp;gt;]&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; text-decoration: underline;&quot;&gt;Basic Types:&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;&lt;br/&gt;&lt;/span&gt;The specifications above make use of a bunch of basic types. These types are: &lt;/p&gt;&lt;p&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;datum &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;weekday&amp;gt; | &amp;lt;day&amp;gt; | &amp;lt;month&amp;gt; | &amp;lt;mday&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;sequence &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;{int} &amp;lt;span&amp;gt; [and {int} &amp;lt;span&amp;gt; …]&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;type &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;datum&amp;gt; | &amp;lt;sequence&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;tpoint &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;date&amp;gt; | &amp;lt;datum&amp;gt; | &amp;lt;year&amp;gt; | &amp;lt;ahead&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/p&gt;&lt;p&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;time &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;[at] {hh[:mm]} | {hh] oclock&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;date &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;{dd-MM-yyyy}&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;weekday &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;{Monday..Sunday}&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;day &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;{01..31}.&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;month &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;{01..12} | {Janurary..December}&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;year &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;yyyy&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;mday &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;{dd-MM}&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;span &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;minute | hour | day | week | month | year&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;ahead &lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;:= &lt;span style=&quot; font-style:italic;&quot;&gt;tomorrow | today&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; text-decoration: underline;&quot;&gt;Custom Formats:&lt;br/&gt;&lt;/span&gt;For many of the formats above, you can customize the keywords to be used in the settings. For the &amp;lt;date&amp;gt; and &amp;lt;mday&amp;gt; formats, you can even define custom formats to be accepted. Check the settings for details on those formats.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="clearButtonEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="importantLabel">
     <property name="text">
      <string>&amp;Important:</string>
     </property>
     <property name="buddy">
      <cstring>importantCheckBox</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QCheckBox" name="importantCheckBox">
     <property name="toolTip">
      <string>Important reminders will show up with additional visual and audio signals to be seens easier</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>23</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="3" column="1">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
  <action name="actionExpression_Syntax">
   <property name="icon">
    <iconset theme="help-whatsthis" resource="remindmewidgets.qrc">
     <normaloff>:/icons/help.ico</normaloff>:/icons/help.ico</iconset>
   </property>
   <property name="text">
    <string>Expression &amp;Syntax</string>
   </property>
   <property name="toolTip">
    <string>Show the expression syntax</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>textLineEdit</tabstop>
  <tabstop>whenLineEdit</tabstop>
  <tabstop>importantCheckBox</tabstop>
 </tabstops>
 <resources>
  <include location="remindmewidgets.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CreateReminderDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CreateReminderDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
